### ISSUE TITLE
[GenericListeners] Member functions not able to read private nor public variables

### ISSUE THREAD

**Issue by RutgerRauws** [igormiktor/arduino-EventManager#13]
Hey Igor,

It's me again. :grimacing:
I've trouble getting the EventManager with GenericListeners to work. I have tried to mimic your WithoutInterrupts example for my own purpose, but it seems my listener and other methods in my MemberFunctionCallable cannot access its public variables. I have attached a minimal working [example](https://github.com/igormiktor/arduino-EventManager/files/306967/example.zip) where I print the public variables to the serial when calling the constructor of the instantiated class, when calling a class' method directly from `setup()` and from the member function listener.

I have defined the following public variables in `ConnectivityManager.h`:

```
byte broadcastIP[4] = {192, 168, 1, 255};
int test = 123;
```

And get the following output:

```
Constructor:
123
192.168.1.255
sendPacket:
130
0.232.232.8
listener:
130
0.208.208.8
```

Clearly, these are just garbage values read from memory, but I don't see the problem in my code. The only difference I seem to have from the example code is the fact that I use separate files for my classes and put my definitions in a separate .cpp file.

What am I doing wrong?

(Sorry if I'm bothering you or bloating the issue page, but I don't think posting this question on the Arduino forums is going to help as this is a very library specific question)

Thanks in advance,
Rutger

**Comment by igormiktor** [comment]
Rutger,

It seems to me you are overrunning the buffer.  "192.168.1.255" is 13 characters long plus a null terminator = 14.  You declared the buffer `char buf[13];` which is too short.  But I'll admit that I don't see exactly how that overflow after you call the constructor in `connectivityManager = new ConnectivityManager();` leads to the other failures without causing other bigger problems. 

In any case, this isn't an EventManger problem:  note that the bug already occurs when you call `connectivityManager->sendPacket();` which is well before the listener gets called.  I suspect that if you stripped everything else out and just had:

```
connectivityManager = new ConnectivityManager();
connectivityManager->sendPacket();
```

The problem would still manifest itself.  But I haven't tried it.

Regards,

Igor

**Comment by RutgerRauws** [comment]
Igor,

I've looked through my code and was wondering where you read that I declared a too short buffer until I realised you meant that the code to test the methods was wrong :see_no_evil:
When starting to make a minimal example, I was actually experiencing a problem with the MemberFunctionCallable which mObj was set to a non-initialized variable. As soon as I fixed that issue, the behavior was unchanged as I was just printing outside the buffer.

Sorry for bothering. I think I've gotten the hang of it!
Rutger
